Dockerfile-alpine

```diff
@@ -0,0 +1,166 @@
+RUN apk add --virtual .build-deps $BUILD_DEPS
+
+# Download the glibc for Alpine .apks, and install
+RUN wget \
+    "$ALPINE_GLIBC_BASE_URL/$ALPINE_GLIBC_PACKAGE_VERSION/$ALPINE_GLIBC_BASE_PACKAGE_FILENAME" \
+    "$ALPINE_GLIBC_BASE_URL/$ALPINE_GLIBC_PACKAGE_VERSION/$ALPINE_GLIBC_BIN_PACKAGE_FILENAME" \
+    "$ALPINE_GLIBC_BASE_URL/$ALPINE_GLIBC_PACKAGE_VERSION/$ALPINE_GLIBC_I18N_PACKAGE_FILENAME"
+RUN apk add \
+    "$ALPINE_GLIBC_BASE_PACKAGE_FILENAME" \
+    "$ALPINE_GLIBC_BIN_PACKAGE_FILENAME" \
+    "$ALPINE_GLIBC_I18N_PACKAGE_FILENAME"
+RUN /usr/glibc-compat/bin/localedef --force --inputfile POSIX --charmap UTF-8 "$LANG" || true
+
+# Cleanup glibc
+RUN rm "/etc/apk/keys/sgerrand.rsa.pub"
+RUN rm "/root/.wget-hsts"
+RUN rm \
+    "$ALPINE_GLIBC_BASE_PACKAGE_FILENAME" \
+    "$ALPINE_GLIBC_BIN_PACKAGE_FILENAME" \
+    "$ALPINE_GLIBC_I18N_PACKAGE_FILENAME"
+
+#-------------------------------------------------------------------------------
+
+# Build libvips
+RUN wget -O- https://github.com/libvips/libvips/releases/download/v${VIPS_VERSION}/vips-${VIPS_VERSION}.tar.gz | tar xzC /tmp
+RUN cd /tmp/vips-${VIPS_VERSION} && \
+    CFLAGS="-g -O3" CXXFLAGS="-D_GLIBCXX_USE_CXX11_ABI=0 -g -O3" \
+    ./configure \
+        --prefix=${VIPS_DIR} \
+        --disable-debug \
+        --disable-dependency-tracking \
+        --disable-introspection \
+        --disable-static \
+        --enable-gtk-doc-html=no \
+        --enable-gtk-doc=no \
+        --enable-pyvips8=no \
+    && make \
+    && make -s install-strip
+
+#-------------------------------------------------------------------------------
+
+# Build Imaginary
+RUN mkdir -p ${GOPATH}/src
+RUN wget -O- https://github.com/h2non/imaginary/archive/v${IMAGINARY_VERSION}.tar.gz | tar xzC ${GOPATH}/src
+RUN cd ${GOPATH}/src/imaginary-${IMAGINARY_VERSION} && \
+    go get -u golang.org/x/net/context && \
+    go get -u github.com/golang/dep/cmd/dep && \
+    go get -u github.com/rs/cors && \
+    go get -u gopkg.in/h2non/filetype.v1 && \
+    go get -u github.com/throttled/throttled && \
+    dep ensure && \
+    go build -ldflags="-s -w" -o $GOPATH/bin/imaginary
+
+#-------------------------------------------------------------------------------
+
+FROM alpine:3.9
+
+ENV PERSISTENT_DEPS \
+    ca-certificates \
+    cairo \
+    expat \
+    fftw \
+    fontconfig \
+    giflib \
+    glib \
+    gobject-introspection \
+    imagemagick \
+    lcms2 \
+    libexif \
+    libgsf \
+    libintl \
+    libjpeg-turbo \
+    libpng \
+    librsvg \
+    libwebp \
+    openexr \
+    openssl \
+    orc \
+    pango \
+    poppler-glib \
+    tiff
+
+RUN apk upgrade --update
+RUN apk add --virtual .persistent-deps $PERSISTENT_DEPS
+RUN echo "@testing http://dl-cdn.alpinelinux.org/alpine/edge/testing" >> /etc/apk/repositories
+RUN apk add libimagequant@testing
+RUN rm -rf /var/cache/apk/*
+
+COPY --from=build /vips/lib/ /usr/local/lib
+COPY --from=build /go/bin/imaginary /usr/bin/imaginary
```

Comment:
Shouldn't we copy over the glibc contents as well (or, reinstall them)? Don't forget `ld.so.conf` that likely lives in `/etc`.

Reply:
Never mind; I missed that you're installing all glibc packages in the runner as well. My guess is that you won't need the `i18n` and `bin` package post generating locale.